charts/kong/values.yaml

```diff
@@ -103,6 +103,22 @@ proxy:
     # Set a nodePort which is available if service type is NodePort
     # nodePort: 32443
 
+  # gRPC/HTTP2 plain-text traffic (disabled by default)
+  grpc:
+    enabled: false
+    servicePort: 9000
+    containerPort: 9000
+    # Set a nodePort which is available if service type is NodePort
+    # nodePort: 32081
+
+  # gRPC/HTTP2 TLS traffic (disabled by default)
+  grpcs:
+    enabled: false
+    servicePort: 9443
+    containerPort: 9443
+    # Set a nodePort which is available if service type is NodePort
+    # nodePort: 32444
```

Comment:
I think rather than introducing a section for `grpc` and `grpcs` here, we should go for something along the lines:

```yaml
additionalPorts:
- servicePort: 9443
  containerPort: 9443
  protocols: # bad name, these are suffixes (https://github.com/Kong/kong/blob/master/kong.conf.default#L195)
  - ssl
  - http2 
  - backlog=N
```

This works for TCP/TLS proxying cases as well.

@rainest has thought about this problem as well and I'm sure he will have more thoughts here.

Reply:
I like this solution! More elegant than what I have. Will make this change.

Reply:
I haven't thought about the values.yaml format much yet (I'd done work on the template end and then realized that we didn't have anything for handling the additional options). I wouldn't want to distinguish between "standard" and "additional" listens, however, since there's not really any distinction between them in the resulting configuration, and it's easier to template if they're both instances of the same type of config.

That, however, would be break with our existing values.yaml format, which does designate specific listens. IMO we should tack parameters (not really a clearer name than the other options, but it's [what NGINX calls them](http://nginx.org/en/docs/http/ngx_http_core_module.html#listen)) onto those existing blocks, e.g.

```
proxy:
  tls:
    enabled: true
    servicePort: 443
    containerPort: 8443
    parameters:
    - http2
    - ssl
    - reuseport
    - backlog=16384
```
We can then defer a more flexible format to some future major release where we do introduce significant breaking changes to the values.yaml format.

That does not allow adding more listens, but I think that's fine for now: for most of our listeners, there's not much reason to add additional ports, since Kong doesn't distinguish between them (you cannot, for example, create a route that is available on one proxy listener but not another). The exception to this is stream listens, which absolutely do function that way, and for which the chart does not have any support currently.

A simpler temporary option (since it may take a while to complete the non-breaking listener rework) would be to just add the [2.0 default parameters](https://github.com/Kong/kong/blob/2.0.0/kong.conf.default#L186) to the [existing proxy_listen helper](https://github.com/helm/charts/blob/master/stable/kong/templates/_helpers.tpl#L60-L69). Although they weren't in the default listener until 2.0, they should be compatible with anything we'd realistically find in the wild.